scene/gui/control.cpp

```diff
@@ -1705,7 +1702,13 @@ Size2 Control::get_end() const {
 
 Point2 Control::get_global_position() const {
 
-	return get_global_transform().get_origin();
+	Transform2D inv;
+
+	if (data.parent_canvas_item) {
```

Comment:
I think this is not a good solution to solve the problem. Getting the parent's transform is not a good solution to retrieve the global transform (mostly because of canvas layers). You should stick to use the node's get_global_transform() and get_transform() for that (not the parent's one).  

I'll try using a simpler solution.

Reply:
My bad, it's done that way in the set_global_position(). So I guess it's ok.

Reply:
Anyway, I would simply replace all of this by a simple :
```
return (get_global_transform() * get_transform().affine_inverse()).xform(get_position())
```
Which is far easier to read.

Reply:
It's definitely nicer. But you need to be more comfortable with matrix multiplication.
I can change it though. shouldn't be touched by end users anyways.

Reply:
Well, you need to be comfortable with transforms if you want to touch this part of the code I guess. So I think is this situation, anything that reduce the number of lines of code is good.